app/web_conn.go

```diff
@@ -53,27 +68,77 @@ type WebConn struct {
 	// It basically acts as the user-space socket buffer, and is used
 	// to resuscitate any messages that might have got lost when the connection is broken.
 	// It is implemented by using a circular buffer to keep it fast.
-	deadQueue        []model.WebSocketMessage
-	deadQueuePointer int // Pointer which indicates the next slot to insert.
-	sessionToken     atomic.Value
-	session          atomic.Value
-	connectionID     atomic.Value
-	endWritePump     chan struct{}
-	pumpFinished     chan struct{}
+	deadQueue []*model.WebSocketEvent
+	// Pointer which indicates the next slot to insert.
+	// It is only to be incremented during writing or clearing the queue.
+	deadQueuePointer int
+	// active indicates whether there is an open websocket connection attached
+	// to this webConn or not.
+	active       bool
+	sessionToken atomic.Value
+	session      atomic.Value
+	connectionID atomic.Value
+	endWritePump chan struct{}
+	pumpFinished chan struct{}
+}
+
+// CheckConnResult indicates whether a connectionID was present in the hub or not.
+// And if so, contains the active and dead queue details.
+type CheckConnResult struct {
```

Comment:
Ahh crap, it needs to be exported because the opentracing package is a sub-package of app, and therefore any struct returned by an app method _must_ be exported. :(

Reply:
Too bad :P